sonar-csharp-plugin/src/main/java/org/sonar/plugins/csharp/CSharpFileCacheSensor.java

```diff
@@ -26,4 +26,9 @@ public class CSharpFileCacheSensor extends AbstractFileCacheSensor {
   public CSharpFileCacheSensor(CSharp cSharp, HashProvider hashProvider) {
     super(cSharp, hashProvider);
   }
+
+  @Override
+  protected String[] additionalSupportedExtensions() {
+    return new String[] { "cshtml" };
+  }
```

Comment:
I assume this is correct because you talked to Costin. But it looks super weird to me. Why do we handle `.cshtml` and `.razor` files differently?

Reply:
Razor files are owned by the C# plugin. Cshtml files are owned by the Web plugin. If I add them to the C# plugin as well then analysis will fail because it will not be able to decide which plugin to use. That's why we need this workaround.